WordPress/gutenberg
# Video: Add aspectRatio block support and utilties

Alternative to #66946
Closes #60911

## What?

This PR adds `dimensions.aspectRatio` support to the video block.

## Why?

In #66946, we are trying to add `aspectRatio` via the `DimensionsTool` component, i.e. as a custom block attribute.

However, the `DimensionsTool` component is intended to be used in a `ToolsPanel`, so this requires refactoring of the `InspectorControls`.

Furthermore, we need to ensure that it is possible to migrate to block support in the future. In that case, I think it would be better to explore whether it is possible to implement it as block support from the start.

## How?

To ensure that styles are applied correctly from both the global styles and block instance, skip serialization and apply the styles to the `video` element inside the block.

## Testing Instructions

- Apply `aspectRatio` to the Video block via theme.json:
  ```json
  {
  	"version": 3,
  	"settings": {
  		"appearanceTools": true,
  		"layout": {
  			"contentSize": "840px",
  			"wideSize": "1100px"
  		}
  	},
  	"styles": {
  		"blocks": {
 			"core/video": {
  				"dimensions": {
  					"aspectRatio": "1/1"
  				}
  			}
  		}
  	}
  }
  ```
- Verify that you can override the aspect ratio via the global styles.
- Verify that you can override the aspect ratio on a block instance.

## Screenshots or screencast 


https://github.com/user-attachments/assets/83b620ae-ba78-43df-b03b-beba39f4e340
